lib/deep_cover/analyser/ruby25_like_branch.rb

```diff
@@ -0,0 +1,214 @@
+# frozen_string_literal: true
+
+require_relative 'subset'
+
+module DeepCover
+  class Analyser::Ruby25LikeBranch < Analyser
+    def self.human_name
+      'Ruby25 branches'
+    end
+    include Analyser::Subset
+    SUBSET_CLASSES = [Node::Case, Node::Csend, Node::If, Node::ShortCircuit,
+                      Node::Until, Node::UntilPost, Node::While, Node::WhilePost,
+                     ].freeze
+
+    def initialize(*args)
+      super
+      @loc_index = 0
+    end
+
+    def results
+      extractor = NodeCoverageExtrator.new
+      each_node.map do |node|
+        extractor.branch_coverage(node)
+      end.to_h
+    end
+
+    class NodeCoverageExtrator < SimpleDelegator
+      def initialize(node = nil)
+        self.node = node
+        @loc_index = 0
+      end
+
+      def node=(value)
+        @delegate_sd_obj = value
+      end
+
+      def node
+        @delegate_sd_obj
```

Comment:
Best use `__getobj__`, or better yet remove the method altogether, I don't know why we'd need it...